workbench-core/accounts/src/services/projectService.ts

```diff
@@ -458,93 +454,104 @@ export default class ProjectService {
   }
 
-  // TODO--implement after dynamic AuthZ
-  // private _generateIdentityPermissionsForProject(projectId: string): IdentityPermissions[] {
-  //   return [
-  //     {
-  //       // create for PA
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#PA`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.CREATE',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'InternalDataset', 'SSHKey', 'User']
-  //     },
-  //     {
-  //       // read for PA
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#PA`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.READ',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'EnvTypeConfig', 'InternalDataset', 'SSHKey', 'User']
-  //     },
-  //     {
-  //       // update for PA
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#PA`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.UPDATE',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'InternalDataset', 'SSHKey']
-  //     },
-  //     {
-  //       // delete for PA
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#PA`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.DELETE',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'InternalDataset', 'SSHKey', 'User']
-  //     },
-  //     {
-  //       // create for Researcher
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#Researcher`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.CREATE',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'SSHKey']
-  //     },
-  //     {
-  //       // read for Researcher
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#Researcher`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.READ',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'EnvTypeConfig', 'InternalDataset', 'SSHKey']
-  //     },
-  //     {
-  //       // update for Researcher
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#Researcher`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.UPDATE',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'SSHKey']
-  //     },
-  //     {
-  //       // delete for Researcher
-  //       identityType: 'GROUP',
-  //       identityId: `${projectId}#Researcher`,
-  //       effect: 'ALLOW',
-  //       action: 'Action.DELETE',
-  //       subject: 'Project',
-  //       subjectId: projectId,
-  //       fields: ['Environment', 'SSHKey']
-  //     }
-  //   ];
-  // }
+  private _generateIdentityPermissionsForProject(projectId: string): IdentityPermission[] {
```

Comment:
nit: can you write a stub for this method? or ask me to since I wrote this method?

Reply:
not sure what you mean by writing a stub for this. its private method so it wouldn't need to be mocked externally if that is the sort of stub you're talking about?

Reply:
doc strings/documentation is what I meant, sorry!